cli/src/cmd/cast/wallet/mod.rs

```diff
@@ -170,6 +190,84 @@ impl WalletSubcommands {
                     }
                 }
             }
+            WalletSubcommands::Import { account_name, keystore_dir, raw_wallet_options } => {
+                // Set up keystore directory
+                let dir = if let Some(path) = keystore_dir {
+                    Path::new(&path).to_path_buf()
+                } else {
+                    Config::foundry_keystores_dir().ok_or_else(|| {
+                        eyre::eyre!("Could not find the default keystore directory.")
+                    })?
+                };
+
+                fs::create_dir_all(&dir)?;
+
+                // check if account exists already
+                let keystore_path = Path::new(&dir).join(&account_name);
+                if keystore_path.exists() {
+                    eyre::bail!("Keystore file already exists at {}", keystore_path.display());
+                }
+
+                // get wallet
+                let wallet: Wallet = raw_wallet_options.into();
+                let wallet = wallet.try_resolve_local_wallet()?.ok_or_else(|| {
+                    eyre::eyre!(
+                        "\
+Did you set a private key or mnemonic?
+Run `cast wallet import --help` and use the corresponding CLI
+flag to set your key via:
+--private-key, --mnemonic-path or --interactive."
+                    )
+                })?;
+
+                let private_key = wallet.signer().to_bytes();
+                let password = rpassword::prompt_password("Enter password: ")?;
+
+                let mut rng = thread_rng();
+                eth_keystore::encrypt_key(
+                    &dir,
+                    &mut rng,
+                    private_key,
+                    &password,
+                    Some(&account_name),
+                )?;
+            }
```

Comment:
I think adding a message that the keystore was saved successfully would be nice! ideally in green

Reply:
Using eyre color, or is there a better alternative? Seems like this clutters autocompletion by recommending all colors for everything that could be printed

Reply:
hey! we have usually use `yansi::Paint` for this—there's a few examples in the codebase, feel free to check them out!

Reply:
Thanks, done now!